src/main/java/edu/kit/compiler/parser/Parser.java

```diff
@@ -40,81 +74,117 @@ public Parser(Lexer lexer) {
      * 
      * @throws ParseException if a syntax error is encountered.
      */
-    public void parse() {
+    public ProgramNode parse() {
+        List<ClassNode> classes = new ArrayList<>();
+        Token programStart = tokenStream.get();
         while (tokenStream.get().getType() == TokenType.Keyword_Class) {
+            Token firstToken = tokenStream.get();
             tokenStream.next();
-            expect(TokenType.Identifier);
+            Token name = expect(TokenType.Identifier);
             expect(TokenType.Operator_BraceL);
-            parseClassMembers();
+            classes.add(parseClassMembers(
+                firstToken.getLine(), firstToken.getColumn(), name.getIntValue().get()
+            ));
             expect(TokenType.Operator_BraceR);
         }
         expect(TokenType.EndOfStream);
+        return new ProgramNode(programStart.getLine(), programStart.getColumn(), classes, false);
     }
 
-    private void parseClassMembers() {
+    private ClassNode parseClassMembers(int line, int column, int className) {
```

Comment:
This method mixes parsing the class members and creating the actual class node. From the name alone, I wouldn't expect it to return an entire class node and also wouldn't understand what the `line`, `column` and `className` parameters are for.

Would it be a solution to split this into `parseClass` and maybe `parseStaticMethod`, `parseDynamicMethod` and `parseField`?

Reply:
`parseDynamicMethod` and `parseField` are a bit hard to separate, but `parseStaticMethod` is separate now and the function parses the whole class.